Maybe-issue on version check in tests

This line checks versions between json file and actual pg version being tested

https://github.com/tds-fdw/tds_fdw/blob/1c38147331b5795348247294bb0f4fbd366def14/tests/lib/tests.py#L54

It fails "on my machine" with an exception :

~~~
[ERROR] '<=' not supported between instances of 'list' and 'str'
Traceback (most recent call last):
  File "postgresql-tests.py", line 127, in <module>
    main()
  File "postgresql-tests.py", line 124, in main
    raise(e)
  File "postgresql-tests.py", line 110, in main
    args.debugging, args.unattended_debugging)
  File "/home/jc/dev/gh/tds_fdw/tests/lib/tests.py", line 125, in run_tests
    if check_ver(conn, req_ver['min'], req_ver['max'], dbtype):
  File "/home/jc/dev/gh/tds_fdw/tests/lib/tests.py", line 54, in check_ver
    if server_ver >= min_ver and (server_ver <= max_ver or min_ver == ''):
TypeError: '<=' not supported between instances of 'list' and 'str'
~~~

If I understand quite correctly this statement, you want to test if current version of PG will be tested ok on the current test.

The point is that if `max_ver` is `''` (that is often), it is evaluated as empty string.

The other point is that `max_ver` should only be tested if defined `max_ver` is define, not `min_ver`.... as far I understand what you want to check.

The final point is that `min_ver` should also be tested as non empty *before* being tested against actual PG version. This should not happen often but it prevent any misuse.

~~~python
if (min_ver ='' or server_ver >= min_ver) and (max_ver == '' or server_ver <= max_ver ):
~~~

Any thought @juliogonzalez  ?

--- a/tests/lib/tests.py
+++ b/tests/lib/tests.py
@@ -51,7 +51,7 @@ def check_ver(conn, min_ver, max_ver, dbtype):
     min_ver = version_to_array(min_ver, dbtype)
     max_ver = version_to_array(max_ver, dbtype)
     server_ver = version_to_array(server_ver, dbtype)
-    if server_ver >= min_ver and (server_ver <= max_ver or min_ver == ''):
+    if server_ver >= min_ver and (server_ver <= max_ver or max_ver == ''):
         return(True)
     else:
         return(False)

